src/model/test/AirConditionerVariableRefrigerantFlowFluidTemperatureControlHR_GTest.cpp

```diff
@@ -0,0 +1,487 @@
+  vrf.addLoadingIndex(loadingIndex4);
+  EXPECT_EQ(4u, vrf.loadingIndexes().size());
+
+  vrf.removeLoadingIndex(loadingIndex3);
+  std::vector<LoadingIndex> loadingIndexes2 = vrf.loadingIndexes();
+  EXPECT_EQ(3u, loadingIndexes2.size());
+  EXPECT_EQ(2, loadingIndexes2[1].compressorSpeed());
+  EXPECT_EQ(evaporativeCapacityMultiplierFunctionofTemperatureCurve2, loadingIndexes2[1].evaporativeCapacityMultiplierFunctionofTemperatureCurve());
+  EXPECT_EQ(compressorPowerMultiplierFunctionofTemperatureCurve2, loadingIndexes2[1].compressorPowerMultiplierFunctionofTemperatureCurve());
+  EXPECT_EQ(4, loadingIndexes2[2].compressorSpeed());
+  EXPECT_EQ(evaporativeCapacityMultiplierFunctionofTemperatureCurve1, loadingIndexes2[2].evaporativeCapacityMultiplierFunctionofTemperatureCurve());
+  EXPECT_EQ(compressorPowerMultiplierFunctionofTemperatureCurve2, loadingIndexes2[2].compressorPowerMultiplierFunctionofTemperatureCurve());
+}
+
+TEST_F(ModelFixture, AirConditionerVariableRefrigerantFlowFluidTemperatureControlHR_Terminals) {
+  Model model;
+
+  AirConditionerVariableRefrigerantFlowFluidTemperatureControlHR vrf(model);
+
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coolingCoil1(model);
+  CoilHeatingDXVariableRefrigerantFlowFluidTemperatureControl heatingCoil1(model);
+  FanVariableVolume fan1(model);
+  ZoneHVACTerminalUnitVariableRefrigerantFlow term1(model, coolingCoil1, heatingCoil1, fan1);
+
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coolingCoil2(model);
+  CoilHeatingDXVariableRefrigerantFlowFluidTemperatureControl heatingCoil2(model);
+  FanVariableVolume fan2(model);
+  ZoneHVACTerminalUnitVariableRefrigerantFlow term2(model, coolingCoil2, heatingCoil2, fan2);
+
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coolingCoil3(model);
+  CoilHeatingDXVariableRefrigerantFlowFluidTemperatureControl heatingCoil3(model);
+  FanVariableVolume fan3(model);
+  ZoneHVACTerminalUnitVariableRefrigerantFlow term3(model, coolingCoil3, heatingCoil3, fan3);
+
+  EXPECT_EQ(0u, vrf.terminals().size());
+
+  vrf.addTerminal(term1);
+  vrf.addTerminal(term2);
+  vrf.addTerminal(term3);
+
+  EXPECT_EQ(3u, vrf.terminals().size());
+
+  vrf.removeTerminal(term2);
+
+  EXPECT_EQ(2u, vrf.terminals().size());
+  EXPECT_EQ(term1, vrf.terminals()[0]);
+  EXPECT_EQ(term3, vrf.terminals()[1]);
+
+  vrf.removeAllTerminals();
+
+  EXPECT_EQ(0u, vrf.terminals().size());
+}
+
+TEST_F(ModelFixture, AirConditionerVariableRefrigerantFlowFluidTemperatureControlHR_Clone) {
+  Model model;
+
+  AirConditionerVariableRefrigerantFlowFluidTemperatureControlHR vrf(model);
+
+  EXPECT_TRUE(vrf.setRefrigerantType("R12"));
+  ZoneHVACTerminalUnitVariableRefrigerantFlow term1(model);
+  vrf.addTerminal(term1);
+  ZoneHVACTerminalUnitVariableRefrigerantFlow term2(model);
+  vrf.addTerminal(term2);
+  EXPECT_EQ(2u, vrf.terminals().size());
+  EXPECT_EQ(3u, vrf.loadingIndexes().size());
```

Comment:
extend this test and similar ones to include a check on the number of curves

Reply:
I extended the tests, but we'll need to follow up on expected values.